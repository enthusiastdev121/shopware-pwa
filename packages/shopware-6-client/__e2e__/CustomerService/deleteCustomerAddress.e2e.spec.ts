import {
  createCustomerAddress,
  deleteCustomerAddress,
  login,
  update,
} from "@shopware-pwa/shopware-6-client";
import { address, name } from "faker";

describe("shopware-6-client - E2E - CustomerService - deleteCustomerAddress", () => {
  beforeEach(() => {
    update({ contextToken: "" });
  });

  it("should delete customer address with given id", async () => {
    await login({ username: "test.e2e@test.pl", password: "password" });
    const customerAddress: any = await createCustomerAddress({
      firstName: name.firstName(),
      lastName: name.lastName(),
      salutationId: "c370eb5cd1df4d4dbcc78f055b693e79",
      zipcode: address.zipCode(),
      city: address.city(),
      street: address.streetName(),
      countryId: "38245a84c3d5425b8bac97fc845b5ddd",
      salutation: "Mr",
    });
    const result = await deleteCustomerAddress(customerAddress.data);
    expect(result).toMatchSnapshot();
  });

  it("should return error deleting not existing customer address", async () => {
    try {
      await login({ username: "test.e2e@test.pl", password: "password" });
      await deleteCustomerAddress("qwa");
      expect("didn't throw an error").toEqual("should throw an error");
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });
});
