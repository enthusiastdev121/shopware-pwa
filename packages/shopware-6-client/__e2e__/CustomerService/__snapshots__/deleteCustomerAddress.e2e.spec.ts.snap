// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopware-6-client - E2E - CustomerService - deleteCustomerAddress should delete customer address with given id 1`] = `undefined`;

exports[`shopware-6-client - E2E - CustomerService - deleteCustomerAddress should return error deleting not existing customer address 1`] = `[Error: Request failed with status code 400]`;
